hardware/raspberrypi/booteeprom.md

```diff
@@ -7,38 +7,99 @@ The Raspberry Pi 4 has an SPI-attached EEPROM (4MBits/512KB), which contains cod
- - The local modifiable state means that OTP bootmode settings will not be required for PXE or USB mass storage boot on the Raspberry Pi 4. There are no user-modifiable OTP bootmode bits on Pi 4.
+ - The local modifiable state means that OTP bootmode settings will not be required for network or USB mass storage boot on the Raspberry Pi 4. There are no user-modifiable OTP bootmode bits on Pi 4.
 
-## PXE and USB Boot
+## Network and USB boot
 
-Support for these additional bootmodes will be added in the future via optional bootloader updates. The current schedule is to release PXE boot first, then USB boot.
+Support for these additional bootmodes will be added in the future via optional bootloader updates. The current schedule is to release network boot first, then USB boot.
 
 ## Is the bootloader working correctly?
 
 To check that the bootloader is working correctly, turn off the power, unplug everything from the Raspberry Pi 4, including the SD card, and then turn the power back on. If the green LED blinks with a repeating pattern then the bootloader is running correctly, and indicating that `start*.elf` has not been found. Any other actions imply that the bootloader is not working correctly and should be reinstalled using `recovery.bin`.
 
-## recovery.bin
+## recovery.bin - rescue image
 
 If the EEPROM needs updating or has somehow become corrupted, it can be reflashed using a fresh SD card with a copy of `recovery.bin` in the first partition of an SD card, formatted to FAT32.
 
 `recovery.bin` is a special utility which runs directly from the SD card and updates the EEPROM — it is not in itself a bootloader. It flashes the green LED rapidly (forever) upon success. Because it's not a bootloader, it won't load `start*.elf`, so once you see the green LED flashing rapidly, just re-insert a regular Raspbian SD card and reboot the Pi.
 
 It can be downloaded from the [raspberrypi.org downloads page](https://www.raspberrypi.org/downloads/).
 
+## Updating the bootloader
+
+We recommend setting up your Pi so that it automatically updates the bootloader: this means you will get new features and bug fixes as they are released. Bootloader updates are performed by the rpi-eeprom package, which installs a service that runs at boot-time to check for critical updates.
+
+```
+sudo apt update
+sudo apt upgrade
+sudo apt install rpi-eeprom
+```
+
+If you wish to control when the updates are applied you can disable the systemd service from running automatically and run 'rpi-eeprom-update' manually (requires sudo)
+
+```
+# Prevent the service from running, this can be run before the package is installed to prevent it ever running automatically.
+sudo systemctl mask rpi-eeprom-update
+
+# Enable it again
+sudo systemctl unmask rpi-eeprom-update
+```
+
+The *FREEZE_VERSION* option the EEPROM config file may be used to indicate that the EEPROM should not be updated on this board. 
```

Comment:
Still missing either "of" or "in" or something... i.e.
The *FREEZE_VERSION* option of the EEPROM config file ...
or
The *FREEZE_VERSION* option in the EEPROM config file ...